cirq-core/cirq/ops/linear_combinations.py

```diff
@@ -570,6 +570,8 @@ def __iadd__(self, other):
         return self
 
     def __add__(self, other):
+        if isinstance(other, (gate_operation.GateOperation)):
```

Comment:
As discussed on the Cirq Cynque, we'll need to make this more specific. 
Namely, the instance check should be about checking whether this is an instance of `identity.IdentityGate`. After this I think we're done with this PR :)

Reply:
but that won't fix the issue in #4087 , the identity there is constructed as `cirq.I(cirq.LineQubit(0))` which has type `cirq.ops.gate_operation.GateOperation` not `cirq.ops.identity.IdentityGate`, so checking the instance type will fail.

Reply:
Oh, yes, I meant instance GateOperation, other.gate is instance of IdentityGate.

Reply:
updated (y).